channel_type.go

```diff
@@ -21,12 +21,43 @@ const (
 	MessageRetentionForever = "infinite"
 )
 
+// ChannelTypeLabel marks a string as a type of channel. The builtin settings are all
+// available as consts.
+type ChannelTypeLabel = string
```

Comment:
It sounds to me `Label` suffix is a bit unnecessary.

Reply:
Agreed. I had a TODO here about finding a better name, sadly the linter is a overly zealous.

`ChannelTypeLabel` is definitely too much, but we can't use `ChannelType` because well that is already a type/struct. So it needs something, label is the best I could come up with. I'm open to suggestions.

Reply:
btw, why don't we use renamed types ?

Reply:
@ferhatelmas sorry I don't know what you mean

Reply:
Oh do you mean like https://play.golang.org/p/RIVqIDAOWGN this seems obvious in retrospect

Reply:
Yes because this way, we don't type safety (not sure if we need) and can't add methods if we need later. Either way, it's an improvement since readability is much better.

For naming, I saw later. Within constraints, sounds good tome.